### PR TITLE
Switch back to MAG_AUTO  + image plotting

### PR DESCRIPTION
**Pull request by virajkaram**
There are some issues with the PSF construction for WINTER images. Switching back to using MAG_AUTO for calculating zeropoints to prevent wildly incorrect results until this is figured out. Also adds image plotting.

### CODE REVIEW

File: mirar/data/utils/plot_image.py [hunk 1]
```diff
@@ -49,9 +52,30 @@ def plot_fits_image(
...
+        # Set title that has latex
+        ax.set_title(rf"{title}", fontsize=10)
+
     ax.set_xlabel("RA")
     ax.set_ylabel("Dec")
-    plot_savepath = savedir / image.header[BASE_NAME_KEY].replace(".fits", ".png")
+    plot_savepath = savedir / image.header[BASE_NAME_KEY].replace(
+        ".fits", f".{plot_format}"
+    )
     logger.info(f"Saving plot to {plot_savepath}")
```
Comment (robertdstein) [inline mirar/data/utils/plot_image.py:79]: debug

File: mirar/processors/utils/image_plotter.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from mirar.data import ImageBatch
+from mirar.data.utils.plot_image import plot_fits_image
+from mirar.paths import get_output_dir
+from mirar.processors.base_processor import BaseImageProcessor
+
+logger = logging.getLogger(__name__)
+
+
+class ImagePlotter(BaseImageProcessor):
+    """Processor to load raw images.
```
Comment (robertdstein) [inline mirar/processors/utils/image_plotter.py:16]: wrong label

File: mirar/processors/utils/image_plotter.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            f"Only pdf and png formats are " f"supported, got {plot_format}."
+        )
+        self.plot_format = plot_format
+        if isinstance(annotate_fields, str):
+            annotate_fields = [annotate_fields]
+        self.annotate_fields = annotate_fields
+
+    def __str__(self):
+        return (
+            f"Processor to plot images and save them in the '{self.output_sub_dir}' "
```
Comment (robertdstein) [inline mirar/processors/utils/image_plotter.py:41]: plot images as {self.format}

File: mirar/processors/utils/image_plotter.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
+"""
+Module to plot images
```
Comment (robertdstein) [inline mirar/processors/utils/image_plotter.py:2]: Please be slightly more explicit somewhere about what the plotting does. Are you basically just plotting the 2D image as a PNG? How do the regions get used?

**Review by robertdstein — CHANGES_REQUESTED** [review]
Looks great, a couple of minor changes. Nate will be happy, since his long-held dream of sending processed images to Slack is closer.

**Comment by robertdstein** [comment]
progress for #109

File: mirar/data/utils/plot_image.py (hunk 1 above)
Comment (virajkaram) [reply mirar/data/utils/plot_image.py:79]: done

File: mirar/processors/utils/image_plotter.py (hunk 2 above)
Comment (virajkaram) [reply mirar/processors/utils/image_plotter.py:16]: done

File: mirar/processors/utils/image_plotter.py (hunk 3 above)
Comment (virajkaram) [reply mirar/processors/utils/image_plotter.py:41]: done

File: mirar/processors/utils/image_plotter.py (hunk 4 above)
Comment (virajkaram) [reply mirar/processors/utils/image_plotter.py]: done, regions don't get used in the processor at all.

File: mirar/processors/utils/image_plotter.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    def __init__(
+        self,
+        output_sub_dir: str = "plots",
+        plot_format: str = "png",
+        annotate_fields: str | list[str] | None = None,
+    ):
+        super().__init__()
+        self.output_sub_dir = output_sub_dir
+        assert plot_format in ["pdf", "png"], (
+            f"Only pdf and png formats are " f"supported, got {plot_format}."
```
Comment (robertdstein) [inline mirar/processors/utils/image_plotter.py:32]: just one minor comment: remove the break in this string

**Review by robertdstein — APPROVED** [review]
One minor change but then please go ahead and merge

**Pull request merged**